src/Messaging/test/RabbitMQ.Test/Host/RabbitMQHostWebApplicationTest.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Steeltoe.Common.Lifecycle;
+using Steeltoe.Connector.RabbitMQ;
+using Steeltoe.Extensions.Configuration.CloudFoundry;
+using Steeltoe.Messaging.RabbitMQ.Configuration;
+using Xunit;
+using RC = RabbitMQ.Client;
+
+namespace Steeltoe.Messaging.RabbitMQ.Host;
+
+public class RabbitMQHostWebApplicationTest
+{
+    [Fact]
+    public void WebAppHostCanBeStarted()
+    {
+        MockRabbitHostedService hostedService;
+
+        WebApplicationBuilder builder = RabbitMQHost.CreateWebApplicationBuilder(null);
+        ;
+        builder.Services.AddSingleton<IHostedService, MockRabbitHostedService>();
+
+        using (WebApplication webApp = builder.Build())
+        {
+            webApp.Start();
+            hostedService = webApp.Services.GetServices<IHostedService>().OfType<MockRabbitHostedService>().First();
+            Assert.NotNull(hostedService);
+            Assert.Equal(1, hostedService.StartCount);
+            Assert.Equal(0, hostedService.StopCount);
+            Assert.Equal(0, hostedService.DisposeCount);
+            webApp.StopAsync().Wait();
+        }
+
+        Assert.Equal(1, hostedService.StartCount);
+        Assert.Equal(1, hostedService.StopCount);
+        Assert.Equal(1, hostedService.DisposeCount);
+    }
+
+    [Fact]
+    public void WebAppHostShouldInitializeServices()
+    {
+        using WebApplication webApp = RabbitMQHost.CreateWebApplicationBuilder().Build();
+        webApp.Start();
+        var lifecycleProcessor = webApp.Services.GetRequiredService<ILifecycleProcessor>();
+        RabbitHostService rabbitHostService = webApp.Services.GetServices<IHostedService>().OfType<RabbitHostService>().First();
+
+        Assert.True(lifecycleProcessor.IsRunning);
+        Assert.NotNull(rabbitHostService);
+        webApp.StopAsync();
+    }
+
+    [Fact]
+    public void WebAppHostShouldAddRabbitOptionsConfiguration()
+    {
+        WebApplicationBuilder builder = RabbitMQHost.CreateWebApplicationBuilder();
+
+        var appSettings = new Dictionary<string, string>
+        {
+            [$"{RabbitOptions.Prefix}:host"] = "ThisIsATest",
+            [$"{RabbitOptions.Prefix}:port"] = "1234",
+            [$"{RabbitOptions.Prefix}:username"] = "TestUser",
+            [$"{RabbitOptions.Prefix}:password"] = "TestPassword"
+        };
+
+        builder.Configuration.AddInMemoryCollection(appSettings);
```

Comment:
```suggestion
        var appSettings = new Dictionary<string, string>
        {
            ["rabbitmq:client:server"] = "ThisIsATest",
            ["rabbitmq:client:port"] = "1234",
            ["rabbitmq:client:username"] = "TestUser",
            ["rabbitmq:client:password"] = "TestPassword"
            //[$"{RabbitOptions.Prefix}:host"] = "ThisIsATest",
            //[$"{RabbitOptions.Prefix}:port"] = "1234",
            //[$"{RabbitOptions.Prefix}:username"] = "TestUser",
            //[$"{RabbitOptions.Prefix}:password"] = "TestPassword"
        };
        WebApplicationBuilder builder = RabbitMQHost.CreateWebApplicationBuilder(configure: configuration => configuration.AddInMemoryCollection(appSettings));

        using WebApplication webApp = builder.Build();
        RabbitOptions rabbitOptions = webApp.Services.GetService<IOptions<RabbitOptions>>()?.Value;

        Assert.NotNull(rabbitOptions);
        Assert.Equal("thisisatest", rabbitOptions.Host);
        Assert.Equal(1234, rabbitOptions.Port);
        Assert.Equal("TestUser", rabbitOptions.Username);
        Assert.Equal("TestPassword", rabbitOptions.Password);
```
Not sure if this is exactly what we're after, but this test does pass for me

Reply:
The change on L73 is notable... without it, the dictionary is not passed to Connectors

Reply:
And then using a RabbitMQHost will not be similar to using generic host, because adding the configuration after (as this test was verifying before modification) no longer works. So it seems we can have one or other but not both. This is a leaky abstraction and I am wondering if we should either move away from it altogether or only have it work for some of the use cases (without connectors). The latter is what I am leaning towards.